src/Runtime/OMTensorList.inc

```diff
@@ -67,33 +64,25 @@ struct OMTensorList {
   OMTensor **_omts; // OMTensor array
 
   int64_t _size; // Number of elements in _omts.
-
-  int64_t _owning; // indicates whether the OMTensorList owns the pointer to
-                   // OMTensor array or not. OMTensorList struct will release
-                   // the memory space referred to by '_omts' upon destruction
-                   // if and only if it owns it.
 };
 
 /* OMTensorList creator */
 OMTensorList *omTensorListCreate(OMTensor **tensors, int64_t n) {
   OMTensorList *list = (OMTensorList *)malloc(sizeof(struct OMTensorList));
   if (!list)
     return NULL;
-  list->_omts = tensors;
   list->_size = n;
-  list->_owning = false;
-  return list;
-}
+  list->_omts = (OMTensor **)malloc(sizeof(OMTensor *) * n);
 
-/* OMTensorList creator with ownership */
-OMTensorList *omTensorListCreateWithOwnership(
-    OMTensor **tensors, int64_t n, int64_t owning) {
-  OMTensorList *list = (OMTensorList *)malloc(sizeof(struct OMTensorList));
-  if (!list)
+  if (!list->_omts) {
+    free(list); // free the previously allocated memory in case of an error
     return NULL;
-  list->_omts = tensors;
-  list->_size = n;
-  list->_owning = owning;
+  }
+
+  for (int64_t i = 0; i < list->_size; i++)
+    // copy the pointers to the OMTensor array
+    // now OMTensorList will always own the tensor array of pointers
+    list->_omts[i] = tensors[i];
```

Comment:
Instead of looping and copying each tensor pointer, `memcpy` should be simpler and faster.

Reply:
Yes I was thinking of using memcpy after talking to Alex but yes both functionality does the same thing. So I just kept this.

Reply:
Debug with the current code, once everything passes, switch to memcpy and verify it still works.

Reply:
Agreed!